File: server/src/main/java/com/fours/onlineschedulerapi/controller/JokeController.java
package com.fours.onlineschedulerapi.controller;

import com.fours.onlineschedulerapi.dto.JokeDto;
import com.fours.onlineschedulerapi.exception.NotAuthorizedException;
import com.fours.onlineschedulerapi.exception.NotFoundException;
import com.fours.onlineschedulerapi.model.Joke;
import com.fours.onlineschedulerapi.model.User;
import com.fours.onlineschedulerapi.service.AuthenticatedUserService;
import com.fours.onlineschedulerapi.service.JokeService;
import com.fours.onlineschedulerapi.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/jokes")
public class JokeController {
    @Autowired
    AuthenticatedUserService authUserService;
    @Autowired
    private UserService userService;
    @Autowired private JokeService jokeService;


    @PostMapping
    public ResponseEntity<?> post(@RequestBody JokeDto joke) {
        try{
            User user = userService.findByName(authUserService.getUsername());
            Joke savedJoke = jokeService.create(joke, user);
            return ResponseEntity.ok(savedJoke);
        }catch (NotFoundException error){
            return new ResponseEntity<>(
                    error.getMessage(), HttpStatus.NOT_FOUND
            );
        }
    }

    @DeleteMapping("/{jokeId}")
    public ResponseEntity<?> delete(@PathVariable Long jokeId) {
        try{
            User user = userService.findByName(authUserService.getUsername());
            jokeService.delete(jokeId, user);
            return new ResponseEntity<>("Joke has been successfully removed.", HttpStatus.OK);
        }catch (NotFoundException error){
            return new ResponseEntity<>(
                    error.getMessage(), HttpStatus.NOT_FOUND
            );
        } catch (NotAuthorizedException error) {
            return new ResponseEntity<>(
                    error.getMessage(), HttpStatus.UNAUTHORIZED
            );
        }
    }
}

File: final project/humorme/src/main/kotlin/com/example/humorme/controller/JokeController.kt
package com.example.humorme.controller

import com.example.humorme.dto.PostJokeDto
import com.example.humorme.model.Joke
import com.example.humorme.service.JokeService
import com.example.humorme.service.UserService
import com.example.humorme.utils.Response
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

//@RestController
//@RequestMapping("/jokes")
@Controller
class JokeController @Autowired constructor(
        private val userService: UserService,
        private val jokeService: JokeService
        ) {

    @PostMapping
    fun doPost(@RequestBody joke: PostJokeDto): Response<Joke>{
        val user = userService.get()
        val newJoke = jokeService.create(joke, user)
        return Response(payload = newJoke)
    }
}